From 8ffc0ab37d9bf7cc46fa9fa27cc1eb7f88f554b1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 4 Dec 2019 23:38:30 -0800
Subject: [PATCH] link libudev for udev_unref and udev_new

These APIs are used in inputlibinput.cpp which is only
compile when HAVE_LIBINPUT is enabled

This shows up when linking with gold linker

../../src/.libs/libegt.so: error: undefined reference to 'udev_new'
../../src/.libs/libegt.so: error: undefined reference to 'udev_unref'
collect2: error: ld returned 1 exit status
Makefile:558: recipe for target 'camera' failed
make[4]: *** [camera] Error 1

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c38a1d0f..e398dedd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -251,7 +251,7 @@ AS_IF([test "x$with_libinput" != "xno"],[
    if test "x${have_libinput}" = xyes; then
       AC_DEFINE(HAVE_LIBINPUT, 1, [Have libinput support])
       LIBEGT_EXTRA_CXXFLAGS="${LIBINPUT_CFLAGS} ${LIBEGT_EXTRA_CXXFLAGS}"
-      LIBEGT_EXTRA_LDFLAGS="${LIBINPUT_LIBS} ${LIBEGT_EXTRA_LDFLAGS}"
+      LIBEGT_EXTRA_LDFLAGS="${LIBINPUT_LIBS} -ludev ${LIBEGT_EXTRA_LDFLAGS}"
    fi
 ])
 AM_CONDITIONAL([HAVE_LIBINPUT], [test "x${have_libinput}" = xyes])
-- 
2.24.0

